### ISSUE TITLE
Please update intel microcode to 0x12B version for MSI Z790 and Z690: Intel's 13 and 14 gen cpu deterioration

### ISSUE THREAD

**Issue by mlario** [Dasharo/dasharo-issues#1079]
### Component

Dasharo firmware

### Device

MSI Pro Z790-P

### Brief summary

Intel microcode update is needed to keep CPU from deteriorating

### How to reproduce

Do you plan to update Intel microcode in Z790 Z690 motherboards to version 0x12B ?

Also could please let us know when updated bios could be available?

Update is needed to keep intel CPUs from thermal degradation and permanent damage as reported before, issue is of high severity: https://www.pcguide.com/news/intel-might-have-finally-dealt-with-those-instability-issues-with-the-0x12b-microcode-update/

Thanks for your support

### Expected behavior

-

### Actual behavior

-

**Comment by zirblazer** [comment]
The developers are aware of this. I have no idea if 0x12B is already available because other vendors have access earlier than public Intel Github releases.

**Comment by SergiiDmytruk** [comment]
> EDIT: I think i understand why It defaulted to 06-97-05 (0x35) microcode, it's because I'm currently using a 12th Gen CPU. If I were to install a 14th or 13th gen CPU it would automatically default to 06-b7-01 (0x129). Am I correct?

Yes, 0x129 targets Raptor Lake only (both 13th and 14th generations) and it's normal to not see it being used for Alder Lake.  Just in case I did check it right now and see Linux reporting `0x129` for 14th gen.

> And @mlario to answer your question, 0x12B hasn't been added to https://github.com/intel/Intel-Linux-Processor-Microcode-Data-Files yet, which is the repo they use to source their microcode.

As was mentioned on Matrix, we might still get 0x12B directly from Intel.  It would not be published separately anywhere though if I understand correctly, meaning that building from source will still use 0x129.

**Comment by miczyg1** [comment]
There are also multiple microcode files for AlderLake that are duplicates o the repo. coreboot uses minimal subset of the files that covers all CPUIDs without taking unnecessary additional space in flash.

**Comment by miczyg1** [comment]
> As was mentioned on Matrix, we might still get 0x12B directly from Intel. It would not be published separately anywhere though if I understand correctly, meaning that building from source will still use 0x129.

I can prepare a properly licensed redistributable file with 0x12B microcode, so let's get it included. I will post a Link to PR soon.

**Comment by miczyg1** [comment]
@SergiiDmytruk you may integrate the blob I have added here: https://github.com/Dasharo/dasharo-blobs/pull/24

Simply change the config to include external microcode file instead of generating from tree, and point to the dasharo-blobs submodule.

**Comment by Firminator** [comment]
From Intel:
> To be clear, The BIOS and microcode updates will not fix processors that already are experiencing instability from Vmin shift. Customers experiencing instability symptoms on 13th and 14th Generation desktop processors should contact their Intel representative, system manufacturer, or place of purchase.

In case your 13th or 14th gen K CPU exposes instabilities due to the voltage issue Intel has extended warranty to 5 years for the following CPUs:
![ExtendedWarranty](https://github.com/user-attachments/assets/7fec0208-9e92-4a28-b6e3-10ffd4c68c82)

Given the criticality of the Intel voltage issues I would be glad if the new 0x12b microcode would be incorporated and released asap at least for Desktop motherboards like Z690/Z790.

Note: mobile 13th and 14th CPUs are not affected, so NovaCustom owners can relax :) 
>Intel® reaffirms that both Intel® Core™ 13th and 14th Gen mobile processors and future client product families – including the codename Lunar Lake and Arrow Lake families - are unaffected by the Vmin Shift Instability issue.

A new out-of-band Dasharo release with the updated microcode definitely double-downs on one of the main reason why we went away from ASUS/MSI/ASROCK/GIGABYTE and their stalling BIOS releases. Sure their BIOS releases seemed to have picked up in speed and frequency in the past 3 years or so, but MSI e.g. has not released 0x12b yet and there is no telling when and if they will do ( https://www.msi.com/Motherboard/PRO-Z690-A-DDR4/support#bios ).

For reference here's the offical statement from Intel: https://community.intel.com/t5/Blogs/Tech-Innovation/Client/Intel-Core-13th-and-14th-Gen-Desktop-Instability-Root-Cause/post/1633239

Intel Recommended BIOS Defaults regarding voltage:
![Intel Recommendations for 13th and 14th K CPUs](https://github.com/user-attachments/assets/d368cd0a-4159-4e8a-a9db-fc079a030363)

**Comment by Firminator** [comment]
Well I'm on Z690 with 12th gen CPU so can't test. Just trying to make a point that this should be released probably asap given the grave implications for K CPUs. Might even consider making this a non-subscriber 'emergency' release (subscriber here). That would really make a point and help adoption further if people see how fast this issue is handled.
But yes Dasharo and the whole crew is out-of-this world. Hats off to the whole team!

**Comment by pietrushnic** [comment]
@BeataZdunczyk @macpijan, shouldn't this issue be included in the release milestone?

**Comment by zirblazer** [comment]
> but MSI e.g. has not released 0x12b yet and there is no telling when and if they will do ( https://www.msi.com/Motherboard/PRO-Z690-A-DDR4/support#bios ).

You jinxed it, they released 7D25v1K with 0x12B microcode a mere 4 days ago.

**Comment by miczyg1** [comment]
@BeataZdunczyk this microcode has been already shipped with the latest v0.9.2 and v1.1.4 releases. THe issue can be closed, unless you want to add it to the milestones.

**Comment by miczyg1** [comment]
@BeataZdunczyk ping.
